### PR TITLE
Adds fixed table layout to table block

### LINKED ISSUE #16045: Block Library: Table: Consider "Fixed width table cells" as default
The table block supports an option for "Fixed width table cells", which changes the appearance of the table so that each column occupies an equal amount of width.

More information about `table-layout: fixed`: https://developer.mozilla.org/en-US/docs/Web/CSS/table-layout

Arguably, this would be the more expected default use-case, and as such we should consider to make this the default behavior.

**Implementation Notes:**

Since this is a change in the default behavior, we should be careful to consider how we want backwards-compatibility to apply. If users have saved content with the current default setting and have expectations around the non-fixed behavior, these expectations should not be broken. Conversely, part of the intent of the defaults implementation of blocks (core blocks specifically) is to allow for revisions to these defaults to more sensible ones in future revisions. 

_Needs Feedback:_ Are we okay in changing the default for existing, saved content?

If existing defaults must be preserved, we should consider how we'd plan to version this, whether that be by introducing a new separate attribute, or some other way to identify the older content for the new default to take effect.

### PR DESCRIPTION
**Pull request by karmatosed**
I have added this to both the editor and also to the front styles. It might be a point for discussion along with should this be done, but also would the editor and theme show it?

Fixes #16045

### CODE REVIEW

**Comment by talldan** [comment]
Hey @karmatosed - what are your thoughts on the existing option for Fixed Table Cells:
Screen Shot 2019-12-03 at 11 36 28 am

Is that something you think could be removed?

**Comment by karmatosed** [comment]
@talldan I do and let me remove that from the PR.

File: packages/block-library/src/table/editor.scss [hunk 1]
```diff
@@ -6,6 +6,7 @@
 		height: auto;
 
 		table {
+			table-layout: fixed;
```
Comment (ZebulanStanphill) [inline packages/block-library/src/table/editor.scss]: Isn't this unnecessary since `style.scss` is loaded in the editor?

Comment (karmatosed) [reply packages/block-library/src/table/editor.scss]: In testing it was for me, feel free to iterate on PR if you don't find it is in testing @ZebulanStanphill

**Review by talldan — COMMENTED** [review]
@karmatosed This looks pretty close, a few things around the tests and deprecations might need a bit of extra work. Let me know if you'd like me to take a look at those few items.

**Comment by karmatosed** [comment]
@talldan what do you think about closing this as it's sat so long? I can always close this and we could start a fresh PR, if easier.

**Comment by annezazu** [comment]
Closing this out -- welcome folks to open a new one to keep it fresh, as said above!

**Pull request closed without merging**